client/state/comments/utils.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import Immutable from 'immutable';
+
+/***
+ * Internal dependencies
+ */
+import {
+	COMBINED_TARGET_TYPE
+} from './constants';
+import {
+	SERIALIZE,
+	DESERIALIZE
+} from '../action-types'
+
+/***
+ * Creates a comment target id, a concatenation of siteId and postId basically
+ * @param {Number} siteId site identification
+ * @param {Number} postId post identification
+ * @returns {String} comment target id
+ */
+export function createCommentTargetId( siteId, postId ) {
+	return `${ siteId }-${ postId }`;
+}
+
+/***
+ * Creates a request id, a concatenation of siteId, postId, and query params basically
+ * @param {Number} siteId site identification
+ * @param {Number} postId post identification
+ * @param {Object} query post identification
+ * @returns {String} request id
+ */
+export function createRequestId( siteId, postId, query ) {
+	const queryKeys = Object.keys( query ).sort();
+	const queryString = queryKeys.map( ( key ) => `${ key }=${ query[key] }` ).join( '-' );
+
+	return `${siteId}-${postId}-${ queryString }`;
+}
+
+/***
+ * Creates a url string from a url template using parameters
+ * @param {String} urlTemplate a template with $var(s) to replace
+ * @param {Object} urlParams parameters for replacement in the template
+ * @param {Object} queryParams query string parameters
+ * @returns {String} url string
+ */
+export function createUrlFromTemplate( urlTemplate, urlParams, queryParams ) {
+	let constructedUrl = urlTemplate;
+
+	Object.keys( urlParams ).forEach( ( key ) => {
+		const regex = new RegExp( '\\$' + key, 'g' );
+		constructedUrl = constructedUrl.replace( regex, urlParams[ key ] );
+	} );
+
+	if ( queryParams ) {
+		constructedUrl += '?' + Object.keys( queryParams )
+				.map( ( param ) => param + '=' + encodeURIComponent( queryParams[ param ] ) )
+				.join( '&' );
+	}
+
+	return constructedUrl;
+}
+
+/***
+ * Similar to redux's combineReducers, combines a reducers to a function that reduces Immutable.Map
+ * @param {Object} reducers key - reducer name, value - reducer function
+ * @param {Function} createId id factory based on action. It should return { type: COMBINED_TARGET_TYPE, id } When the factory returns null, reducer skips processing
+ * @returns {Function} combined reducer function
+ */
+export function combineReducersWithId( reducers, createId ) {
```

Comment:
I'm not really sure what you mean, I'll try explain more where I'm coming from.

Let's assume that we want the following structure for comments state:
```javascript
comments[ `${ site_ID }_${ post_ID }` ] = { tree, totalCommentsCount, ... }
```
Where `comments` is `Immutable.Map`.

A reducer for that state structure will look something like that (psuedo-code):
```javascript
// comments.js
export default function reducer( state = Immutable.Map(), action ) {
	if ( ! action.siteId || ! action.postId ) {
		return state;
	}	
	const id = action.siteId + '_' + action.postId;

	// ensure we have a map for the id
	let newState = state;
	if ( ! newState.get( id ) ) {
		newState = newState.set( id, Immutable.Map() );
	}
	
	switch ( action.type ) {
			case COMMENTS_RECEIVE:
				return newState.setIn( [ id, 'tree' ], tree( action ) );
			case COMMENTS_COUNT_RECEIVE:
				return newState.setIn( [ id, 'totalCommentsCount' ], totalCommentsCount( action ) );
			
			... more action(s)...
	}
	
	// nothing changed, return original
	return state;
}
```

Notice the `tree` and `totalCommentsCount` functions, they will actually perform the state change,
as we will have more actions, we'll have to add them to that reducer, as a `CASE`, and create
a function that will handle it. Other option ofcurse is have the functions inline, in `totalCommentsCount` case
that doesn't really matter as it's really tiny, but at tree's case it does.
IMO, it also creates a long reducer function that's harder to follow, and I prefer to avoid that if possible.

`combineReducersWithId()` is simply a general version of the above reducer, it has nothing special about it,
it just creates a `comments( state, action ) = newState` function.

You've mentioned you 
> prefer to use the standard reducer

I'm not really sure what you mean, did you mean you'd like to use a _long_ reducer function as an the example above?

Or just use many shorter reducer(s) that create the form of:
```javascript
comments = Map { 
    tree: Map {
        [ `${ site_ID }_${ post_ID }` ]: { ...some comments tree... }
    },
    totalCommentsCount: Map {
       [ `${ site_ID }_${ post_ID }` ]: 1024
    },
    ...more props...
}
```

Or another option?

Thank you for your time reviewing this. =)